components/social/org.wso2.carbon.social.sql/src/main/java/org/wso2/carbon/social/sql/SQLActivityBrowser.java

```diff
@@ -588,4 +637,38 @@ public JsonObject getPopularAssets(String type, String tenantId, int limit,
 
 	}
 
+	/**
+	 * Check whether the given user has already reviewed on the given targetId (AssetType + UUID)
+	 * @param targetId AssetType + UUID delimited by colon
+	 * @param userId Username with tenant domain i:e admin@carbon.super
+	 * @return Boolean whether user has already reviewed or not
+	 */
+    public boolean isReviewed(String targetId, String userId) throws SocialActivityException {
+        boolean reviewed = false;
+        Connection connection = null;
+        PreparedStatement statement;
+        ResultSet resultSet;
+        String errorMsg = "Unable to retrieve comments for the user : " + userId;
+
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("Executing: " + IS_REVIEWED_SELECT_SQL);
+            }
+            connection = DSConnection.getConnection();
+            statement = connection.prepareStatement(IS_REVIEWED_SELECT_SQL);
+            statement.setString(1, targetId);
+            statement.setString(2, userId);
+            resultSet = statement.executeQuery();
+            reviewed = resultSet.next();
+            resultSet.close();
+        } catch (SQLException | DataSourceException e) {
+            String message = errorMsg + e.getMessage();
+            log.error(message, e);
+            throw new SocialActivityException(message, e);
```

Comment:
no need to add e.getMessage(). it is already there in e. just pass errorMsg.
no need to log the message, since we throw the same.